app/core/auth-options.ts

```diff
@@ -206,6 +270,11 @@ export const authOptions: AuthOptions = {
       const loweremail = email.toLowerCase();
       const lastSeen = new Date();
 
+      await prisma.user.update({
```

Comment:
there is `upsert` command just below; why need two database calls that does the same? and this won't work if the user is the new.